### PR TITLE
GH-37231: [MATLAB] Add `arrow.type.Time32Type` class and `arrow.time32` construction function

### LINKED ISSUE #37231: [MATLAB] Add `arrow.type.Time32Type` class and `arrow.time32` construction function
### Describe the enhancement requested

To add support for `arrow.array.Time32Array`, we will need to add support for a `arrow.type.Time32Type` class and associated `arrow.time32` construction function to the MATLAB interface.

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by kevingurney**
### Rationale for this change

To support the addition of `arrow.array.Time32Array`, this pull request adds a new `arrow.type.Time32Type` class and associated `arrow.time32` construction function to the MATLAB interface.

### What changes are included in this PR?

1. New `arrow.type.Time32Type` class.
2. New `arrow.time32` construction function that returns an `arrow.type.Time32Type` instance.

**Example**

```matlab
>> type = arrow.time32(TimeUnit="Millisecond")

type = 

  Time32Type with properties:

          ID: Time32
    TimeUnit: Millisecond

>> class(type)

ans =

    'arrow.type.Time32Type'
```

### Are these changes tested?

Yes.

1. Added new tests for `Time32` type ID enumeration to `tID`.
2. Added a test class `tTimeUnit` for the new validation function `arrow.internal.validate.temporal.timeUnit`.
4. Added a new test class `tTime32Type`.

### Are there any user-facing changes?

Yes.

There are two new user-facing APIs:

1. `arrow.time32` construction function
2. `arrow.type.Time32Type` class

### Future Directions:

1. #37232
2. #37229
3.  #37230
4. #37251

### Notes

1. Thank you @sgilmore10 for your help with this pull request!
* Closes: #37231

### CODE REVIEW

File: matlab/src/cpp/arrow/matlab/type/proxy/time32_type.cc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    libmexclass::proxy::MakeResult Time32Type::make(const libmexclass::proxy::FunctionArguments& constructor_arguments) {
+        namespace mda = ::matlab::data;
+
+        using Time32TypeProxy = arrow::matlab::type::proxy::Time32Type;
+
+        mda::StructArray opts = constructor_arguments[0];
+
+        // Get the mxArray from constructor arguments
```
Comment (sgilmore10) [inline matlab/src/cpp/arrow/matlab/type/proxy/time32_type.cc:36]: Can remove this comment probably?

File: matlab/test/arrow/type/tTime32Type.m [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        end
+
+        function TimeUnitNoSetter(testCase)
+            % Verify that an error is thrown when trying to set the value
+            % of the TimeUnit property.
+            schema = arrow.time32(TimeUnit="Millisecond");
+            testCase.verifyError(@() setfield(schema, "TimeUnit", "Second"), "MATLAB:class:SetProhibited");
+        end
+
+        function BitWidthNoSetter(testCase)
```
Comment (sgilmore10) [inline matlab/test/arrow/type/tTime32Type.m:114]: Can you create an issue to add `BitWidthNoSetter`, `IDNoSetter`,  and`NumFieldsNoSetter` test cases to the parent test class?

**Review by sgilmore10 — APPROVED** [review]

File: matlab/test/arrow/type/tTime32Type.m (hunk 2 above)
Comment (kevingurney) [reply matlab/test/arrow/type/tTime32Type.m:114]: That's a good point. Created issue #37253.

Comment (kevingurney) [reply matlab/test/arrow/type/tTime32Type.m:114]: Marking this as resolved.

File: matlab/src/cpp/arrow/matlab/type/proxy/time32_type.cc (hunk 1 above)
Comment (kevingurney) [reply matlab/src/cpp/arrow/matlab/type/proxy/time32_type.cc:36]: Good point.

Comment (kevingurney) [reply matlab/src/cpp/arrow/matlab/type/proxy/time32_type.cc]: Fixed.

**Comment by kevingurney** [comment]
+1

**Pull request merged**